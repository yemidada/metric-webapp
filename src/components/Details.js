import { faCircleRight } from '@fortawesome/free-regular-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import PropTypes from 'prop-types';

const Details = ({ title, answer }) => (
  <div className="p-3 page-details d-flex justify-content-between align-items-center">
    <div>
      {title}
    </div>
    <div>
      <span className="answer">{ answer }</span>
      <FontAwesomeIcon icon={faCircleRight} />
    </div>
  </div>
);

Details.defaultProps = {
  answer: PropTypes.string,
  title: PropTypes.string,
};

Details.propTypes = {
  answer: PropTypes.string,
  title: PropTypes.string,
};

export default Details;
