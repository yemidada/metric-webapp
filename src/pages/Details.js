import { useLocation } from "react-router-dom";
import Feature from "../components/Feature";
import Title from "../components/Title";
import Details from "../components/Details";

const Home = () => {
  const location = useLocation();
  const data = location.state?.data;
  return (
    <>
      <Feature item={data} className="feature"/>
      <Title title="DETAIL OF HAIRSTYLE" />
      <Details title="Name" answer={data && data.name ? data.name : ''} />
      <Details title="Owner" answer={data && data.owned ? data.owned : ''}  />
      <Details title="Patch" answer={data && data.patch ? data.patch : ''}  />

    </>
  )};
  export default Home;