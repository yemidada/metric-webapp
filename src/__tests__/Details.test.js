import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Details from '../pages/Details';

describe('Homepage component test', () => {
  it('should render Home page correctly', () => {
    const details = render(
      <Router>
        <Provider store={store}>
          <Details />
        </Provider>
      </Router>,
    );

    expect(details).toMatchSnapshot();
  });
});