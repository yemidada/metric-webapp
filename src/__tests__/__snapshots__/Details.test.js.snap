// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage component test should render Home page correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="page-feature"
      >
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col-6"
            />
            <div
              class="col-6 item-detail"
            >
              <p />
              <p
                class="value"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="page-title"
      >
        <div
          class="container"
        >
          DETAIL OF HAIRSTYLE
        </div>
      </div>
      <div
        class="p-3 page-details d-flex justify-content-between align-items-center"
      >
        <div>
          Name
        </div>
        <div>
          <span
            class="answer"
          />
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-right "
            data-icon="circle-right"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M464 256A208 208 0 1 1 48 256a208 208 0 1 1 416 0zM0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM294.6 135.1c-4.2-4.5-10.1-7.1-16.3-7.1C266 128 256 138 256 150.3V208H160c-17.7 0-32 14.3-32 32v32c0 17.7 14.3 32 32 32h96v57.7c0 12.3 10 22.3 22.3 22.3c6.2 0 12.1-2.6 16.3-7.1l99.9-107.1c3.5-3.8 5.5-8.7 5.5-13.8s-2-10.1-5.5-13.8L294.6 135.1z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
      <div
        class="p-3 page-details d-flex justify-content-between align-items-center"
      >
        <div>
          Owner
        </div>
        <div>
          <span
            class="answer"
          />
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-right "
            data-icon="circle-right"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M464 256A208 208 0 1 1 48 256a208 208 0 1 1 416 0zM0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM294.6 135.1c-4.2-4.5-10.1-7.1-16.3-7.1C266 128 256 138 256 150.3V208H160c-17.7 0-32 14.3-32 32v32c0 17.7 14.3 32 32 32h96v57.7c0 12.3 10 22.3 22.3 22.3c6.2 0 12.1-2.6 16.3-7.1l99.9-107.1c3.5-3.8 5.5-8.7 5.5-13.8s-2-10.1-5.5-13.8L294.6 135.1z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
      <div
        class="p-3 page-details d-flex justify-content-between align-items-center"
      >
        <div>
          Patch
        </div>
        <div>
          <span
            class="answer"
          />
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-right "
            data-icon="circle-right"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M464 256A208 208 0 1 1 48 256a208 208 0 1 1 416 0zM0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM294.6 135.1c-4.2-4.5-10.1-7.1-16.3-7.1C266 128 256 138 256 150.3V208H160c-17.7 0-32 14.3-32 32v32c0 17.7 14.3 32 32 32h96v57.7c0 12.3 10 22.3 22.3 22.3c6.2 0 12.1-2.6 16.3-7.1l99.9-107.1c3.5-3.8 5.5-8.7 5.5-13.8s-2-10.1-5.5-13.8L294.6 135.1z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="page-feature"
    >
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="col-6"
          />
          <div
            class="col-6 item-detail"
          >
            <p />
            <p
              class="value"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="page-title"
    >
      <div
        class="container"
      >
        DETAIL OF HAIRSTYLE
      </div>
    </div>
    <div
      class="p-3 page-details d-flex justify-content-between align-items-center"
    >
      <div>
        Name
      </div>
      <div>
        <span
          class="answer"
        />
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-circle-right "
          data-icon="circle-right"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 256A208 208 0 1 1 48 256a208 208 0 1 1 416 0zM0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM294.6 135.1c-4.2-4.5-10.1-7.1-16.3-7.1C266 128 256 138 256 150.3V208H160c-17.7 0-32 14.3-32 32v32c0 17.7 14.3 32 32 32h96v57.7c0 12.3 10 22.3 22.3 22.3c6.2 0 12.1-2.6 16.3-7.1l99.9-107.1c3.5-3.8 5.5-8.7 5.5-13.8s-2-10.1-5.5-13.8L294.6 135.1z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div
      class="p-3 page-details d-flex justify-content-between align-items-center"
    >
      <div>
        Owner
      </div>
      <div>
        <span
          class="answer"
        />
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-circle-right "
          data-icon="circle-right"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 256A208 208 0 1 1 48 256a208 208 0 1 1 416 0zM0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM294.6 135.1c-4.2-4.5-10.1-7.1-16.3-7.1C266 128 256 138 256 150.3V208H160c-17.7 0-32 14.3-32 32v32c0 17.7 14.3 32 32 32h96v57.7c0 12.3 10 22.3 22.3 22.3c6.2 0 12.1-2.6 16.3-7.1l99.9-107.1c3.5-3.8 5.5-8.7 5.5-13.8s-2-10.1-5.5-13.8L294.6 135.1z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div
      class="p-3 page-details d-flex justify-content-between align-items-center"
    >
      <div>
        Patch
      </div>
      <div>
        <span
          class="answer"
        />
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-circle-right "
          data-icon="circle-right"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 256A208 208 0 1 1 48 256a208 208 0 1 1 416 0zM0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM294.6 135.1c-4.2-4.5-10.1-7.1-16.3-7.1C266 128 256 138 256 150.3V208H160c-17.7 0-32 14.3-32 32v32c0 17.7 14.3 32 32 32h96v57.7c0 12.3 10 22.3 22.3 22.3c6.2 0 12.1-2.6 16.3-7.1l99.9-107.1c3.5-3.8 5.5-8.7 5.5-13.8s-2-10.1-5.5-13.8L294.6 135.1z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
